pdata/pprofile/attributes.go

```diff
@@ -31,38 +31,79 @@ func FromAttributeIndices(table AttributeTableSlice, record attributable) pcommo
 	return m
 }
 
-// AddAttribute updates an AttributeTable and a record's AttributeIndices to
-// add a new attribute.
-// The record can by any struct that implements an `AttributeIndices` method.
-func AddAttribute(table AttributeTableSlice, record attributable, key string, value pcommon.Value) error {
```

Comment:
Given that `AddAttribute` is released, please mark it as deprecated for now and remove in a future version. See https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/coding-guidelines.md#api-breaking-changes